contrib/linux/sensors/file_watch_sensor.py

```diff
@@ -42,6 +44,9 @@ def add_trigger(self, trigger):
         if not self._trigger:
             raise Exception('Trigger %s did not contain a ref.' % trigger)
 
+        # Wait a bit to avoid initialization race in logshipper library
+        eventlet.sleep(1.0)
```

Comment:
There is an edge case in logshipper library.

If a trigger is added and ``add_file`` is called before the tailer is initialized, an exception will be thrown and the sensor will die / crash (this can happen because ``add_trigger`` method is called asynchronously and it could be called before ``run()`` and tailer fully initialized).

Reply:
:(

Reply:
Yeah, it's not ideal.

I could make it 100% deterministic by shuffling around the code (set ``initialized`` or similar variable to True inside ``run()`` and only proceed inside ``add_handler`` when that variable is True).

I think that's too complicated though since sleep seems to work just as fine (and it's one line of code).

It also only affects a single sensor right now so there is no point in handling that inside sensor container / base sensor class code.